src/long.ts

```diff
@@ -246,13 +247,43 @@ export class Long extends BSONValue {
   }
 
   /**
+   * @internal
+   * Returns false for an string that contains invalid characters for its radix, else returns the original string.
+   * @param str - The textual representation of the Long
+   * @param radix - The radix in which the text is written (2-36), defaults to 10
+   */
+  static validateStringCharacters(str: string, radix?: number): false | string {
+    radix = radix ?? 10;
+
+    let regexInputString = '';
+    if (radix <= 10) {
+      regexInputString = `[^-0-${radix - 1}+]`;
+    } else {
+      const validCharRangeEnd = String.fromCharCode('a'.charCodeAt(0) + (radix - 11));
+      regexInputString = `[^-0-9+(a-${validCharRangeEnd})]`;
+    }
```

Comment:
regardless of the implementation chosen, can we add comments explaining what this regex is looking for?  regexes tend to be pretty opaque.

Could we consolidate these branches with something like this?
```suggestion
const validDigits = '0123456789abcdefghijklmnopqrstuvwxyz`.slice(0, radix);
const regex = `[^-+${validDigits}`];
...
```